beacon/weakcoin/weak_coin.go

```diff
@@ -222,6 +235,7 @@ func (wc *WeakCoin) prepareProposal(epoch types.EpochID, round types.RoundID) ([
 				Round:     round,
 				Unit:      unit,
 				MinerPK:   wc.signer.PublicKey().Bytes(),
+				Nonce:     nonce,
```

Comment:
same here. this is the gossiped message. not the vrf message. don't add nonce here.